File: ios/Classes/OnBoardIdUiBridge.swift
import Flutter
import NetkiSDK

class OnBoardIdUiBridge {

    func handle(call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "applyTheme":
            applyTheme(call: call, result: result)
        default:
            result(FlutterMethodNotImplemented)
        }
    }

    private func applyTheme(call: FlutterMethodCall, result: @escaping FlutterResult) {
        guard let args = call.arguments as? [String: Any],
              let themeMap = args["theme"] as? [String: Any] else {
            result(FlutterError(code: "INVALID_ARGUMENTS", message: "Theme is required", details: nil))
            return
        }

        let theme = Mappers.mapToOnBoardIdTheme(themeMap)
        OnBoardIdUiV2.applyTheme(theme)
        result(nil)
    }
}

File: android/src/main/kotlin/com/netki/netki_sdk/OnBoardIdUiBridge.kt
package com.netki.netki_sdk

import com.netki.OnBoardIdUiV2
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel.Result

class OnBoardIdUiBridge {

    fun handle(call: MethodCall, result: Result) {
        when (call.method) {
            "applyTheme" -> applyTheme(call, result)
            else -> result.notImplemented()
        }
    }

    private fun applyTheme(call: MethodCall, result: Result) {
        try {
            @Suppress("UNCHECKED_CAST")
            val args = call.arguments as? Map<String, Any?>
            if (args == null) {
                result.error("INVALID_ARGUMENT", "Arguments are required", null)
                return
            }

            @Suppress("UNCHECKED_CAST")
            val themeMap = args["theme"] as? Map<String, Any?>
            if (themeMap == null) {
                result.error("INVALID_ARGUMENT", "Theme map is required", null)
                return
            }

            val theme = Mappers.mapToOnBoardIdTheme(themeMap)
            OnBoardIdUiV2.applyTheme(theme)
            result.success(null)
        } catch (e: Exception) {
            result.error("THEME_ERROR", e.message, e.stackTraceToString())
        }
    }
}
